src/native/libs/System.Globalization.Native/pal_collation.c

```diff
@@ -335,7 +335,10 @@ static UCollator* CloneCollatorWithOptions(const UCollator* pCollator, int32_t o
 #if HAVE_UCOL_CLONE
         pClonedCollator = ucol_clone(pCollator, pErr);
 #else
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
         pClonedCollator = ucol_safeClone(pCollator, NULL, NULL, pErr);
+#pragma clang diagnostic pop
```

Comment:
The compilation uses Homebrew ICU headers which are newer (>=71) than the shipped ICU.